osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableSwell.cs

```diff
@@ -195,10 +193,10 @@ protected override void Update()
             X = Math.Max(0, X);
 
             double t = Math.Min(HitObject.StartTime, Time.Current);
-            if (t == HitObject.StartTime && !hasStarted)
+            if (t == HitObject.StartTime && !hasProxied)
             {
-                OnStart?.Invoke();
-                hasStarted = true;
+                ProxyContent();
+                hasProxied = true;
```

Comment:
i think we can move this `hasProxied` state to `DrawableTaikoHitObject` and return early (seems safer than relying on each caller doing this).